### ISSUE TITLE
React Native Web: Cannot use import statement outside a module. AccessibilityInfo

### ISSUE THREAD

**Issue by FrancisLeigh** [martpie/next-transpile-modules#127]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE
- [ ] I WILL GIVE 10$ TO CHARITY IF MY SOLUTION WAS ACTUALLY IN THE FAQ

**Are you trying to transpile a local package or an npm package?**
npm[ package](https://www.npmjs.com/package/maisonsport-common-ui) 

**Describe the bug**
With a fresh `next-with-react-native-web` Project I install the npm library.
I run `npm run dev`. the build compiles successfully but when I load localhost:3000 I am greeted with 
```
import typeof AccessibilityInfo from './Libraries/Components/AccessibilityInfo/AccessibilityInfo';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**To Reproduce**
1. `next with-react-native-web` project
2. `npm i maisonsport-common-ui`
3. `npm run dev` 
4. If initially it works then reload the browser and witness failure. 

**Expected behavior**
Page should render fine with components from npm package.

**Setup**

- Next.js version: latest (installed today)
- `"next-transpile-modules": "^4.1.0",`
- Node.js version: `v14.8.0`
- `npm` version: `6.14.7`
- Operating System: MacOS
- Webpack 4: `4.44.1`

**Additional context**
See an [issue I raised with next](https://github.com/vercel/next.js/issues/18029) I added a screen recording of some strange behavior as well which might be useful.

## next.config.js
```javascript
const withTM = require('next-transpile-modules')(['maisonsport-common-ui']);

module.exports = withTM({
  webpack: (config) => {
    config.resolve.alias = {
      ...(config.resolve.alias || {}),
      // Transform all direct `react-native` imports to `react-native-web`
      'react-native$': 'react-native-web',
    }
    config.resolve.extensions = [
      '.web.ts',
      '.web.tsx',
      '.ts',
      '.tsx',
      '.web.js',
      '.web.jsx',
      '.js',
      '.jsx',
      ...config.resolve.extensions,
    ]

    return config;
  },
})

```

## babel.config.js
```javascript
module.exports = {
  presets: ['next/babel'],
  plugins: [
    ['react-native-web', { commonjs: true }],
    [
      '@babel/plugin-transform-runtime',
      {
        absoluteRuntime: false,
        corejs: false,
        helpers: true,
        regenerator: true,
        useESModules: true,
        version: '7.0.0-beta.0',
      },
    ]
  ],
}
```

**Comment by nandorojo** [comment]
Try using the `@expo/next-adapter`. I use this for RNW and it works well.

**Comment by nandorojo** [comment]
This could also happen if you use any packages that import React Native but aren't in the transpile modules array.

**Comment by MoOx** [comment]
@FrancisLeigh You need to add all react-native package to the transpiled list, it's that simple. Like this it's easy:

https://github.com/MoOx/moox.io/blob/db88a119c43deafab15afa2d11deac4b43700238/next.config.js#L7-L9

**Comment by MoOx** [comment]
@martpie config problem, this issue can be closed, no issue with RNW, I use it successfully on several next projects.

**Comment by nandorojo** [comment]
Should we transpile RN, or packages that consume it? Is there a benefit to one vs the other?

I'm trying to reduce the time it takes to transpile, so I wonder if there are any obvious reasons for either option.

Right now, I transpile a few libs that import RN, but not RN itself. I also use expo's Nextjs webpack config, so maybe they're doing that for me(?)

**Comment by MoOx** [comment]
All the RN ecosystem is published untranspiled, so you need to transpire all RN packages. And also RNW. You basically need to transpire all modules that raise syntax error on import if I am correct.

**Comment by rxb** [comment]
Does the need to transpile RN and RNW start when you have at least one module that needs transpiling? I'm using RNW in my nextjs project with no transpiling.

**Comment by MoOx** [comment]
What you might need to transpile are RN modules, not RNW itself.

**Comment by KeitelDOG** [comment]
@MoOx I had already done it with React-Native-Web and webpack, now thanks to your config, I combined it and make it work with NextJS. You're a hero!
